Recursion/TowerHanoi.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+@Author Davide Spataro www.davidespataro.it
+
+Tower Of Hanoi
+-------------------
+Solves the tower of hanoi problem (https://en.wikipedia.org/wiki/Tower_of_Hanoi)
+using a recursive approach.
+
+
+Time complexity
+---------------
+O(2^n),where n is the number of disks
+
+Space complexity
+----------------
+O(n), where n is the number of disks
+(not considering the stack space for the required by recursion).
+*/
+
+
+#include <iostream>
+#include <array>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::array;
+using std::cin;
+using std::vector;
+
+/*
+  Class HanoiTowerSolver. it exposes a simple interface for solving
+  hanoi tower problem.
+*/
+class HanoiTowerSolver {
```

Comment:
I think that using a solver class is unnecessary. It was fine the way it was before:
- A function that recursively solves the Tower of Hanoi problem, and after each move calls:
- Another function that draws the current status of the 3 piles.

This way it's much simple to understand. The initialisation etc. can be done in `main()`.